johnzon-maven-plugin/pom.xml

```diff
@@ -119,4 +119,86 @@
       </plugin>
     </plugins>
   </build>
+
+  <profiles>
+    <profile>
+      <id>jdk9+</id>
+      <activation>
+        <jdk>[9,)</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
```

Comment:
let's keep this PR about OSGi maybe

side note: if we compile the generated classes, I'd be to do it in the test doing the generation with ToolProvider or a plain System.getProperty(java.home) + javac lookup and not a maven plugin to keep it consistent and localized. For record java.version can be tested but releases will be done with java 8 (or 11) so not sure it will help to prevent much in the future at community level.

Reply:
done